### PR TITLE
Gsettings refactoring

### PR DESCRIPTION
**Pull request by joroca**
- Compliant Chef 12.5.0
- Migrates old LWRP gsettings to new Custom Resource desktop_gsettings
- Invokes resource:
  desktop_gsettings "org.nemo.preferences" do
    key "default-folder-viewer"
    value user.default_folder_viewer
    user username
   action: nothing
  end.run_action(:set)

### CODE REVIEW

File: providers/file_browser.rb [hunk 1]
```diff
@@ -20,58 +20,53 @@
...
-            username username
-            provider "gecos_ws_mgmt_gsettings"
             action :nothing
           end.run_action(:set)
         end
 
         #show_hidden_files
         if !user.show_hidden_files.empty? and !user.show_hidden_files.nil? 
-          gecos_ws_mgmt_desktop_setting "show-hidden-files" do
+          desktop_gsettings "org.nemo.preferences" do
```
Comment (alfem) [inline providers/file_browser.rb]: Avoid reusing a resource name ("org.nemo.preferences"). Append the key name to this resource.

File: providers/file_browser.rb [hunk 2]
```diff
@@ -20,58 +20,53 @@
...
-            username username
-            provider "gecos_ws_mgmt_gsettings"
             action :nothing
           end.run_action(:set)
         end
      
         #show_search_icon_toolbar
         if !user.show_search_icon_toolbar.empty? and !user.show_search_icon_toolbar.nil? 
-          gecos_ws_mgmt_desktop_setting "show-search-icon-toolbar" do
+          desktop_gsettings "org.nemo.preferences" do
```
Comment (alfem) [inline providers/file_browser.rb]: Avoid reusing a resource name ("org.nemo.preferences"). Append the key name to this resource.

File: providers/file_browser.rb [hunk 3]
```diff
@@ -20,58 +20,53 @@
...
-            username username
-            provider "gecos_ws_mgmt_gsettings"
             action :nothing
           end.run_action(:set)
         end
 
         #click_policy
         if !user.click_policy.empty? and !user.click_policy.nil? 
-          gecos_ws_mgmt_desktop_setting "click-policy" do
+          desktop_gsettings "org.nemo.preferences" do
```
Comment (alfem) [inline providers/file_browser.rb]: Avoid reusing a resource name ("org.nemo.preferences"). Append the key name to this resource.

File: providers/file_browser.rb [hunk 4]
```diff
@@ -20,58 +20,53 @@
...
-            username username
-            provider "gecos_ws_mgmt_gsettings"
             action :nothing
           end.run_action(:set)
         end
 
         #confirm_trash
         if !user.confirm_trash.empty? and !user.confirm_trash.nil? 
-          gecos_ws_mgmt_desktop_setting "confirm-trash" do
+          desktop_gsettings "org.nemo.preferences" do
```
Comment (alfem) [inline providers/file_browser.rb]: Avoid reusing a resource name ("org.nemo.preferences"). Append the key name to this resource.

File: providers/screensaver.rb [hunk 5]
```diff
@@ -31,39 +31,31 @@
   ## Distinguir entre sesion Cinnamon y LXDE
 
   #     session = node["desktop_session"] 
-        gecos_ws_mgmt_desktop_setting "idle-activation-enabled" do
-          type "string"
+        desktop_gsettings "org.cinnamon.desktop.screensaver" do
```
Comment (alfem) [inline providers/screensaver.rb]: Avoid reusing a resource name ("org.cinnamon.desktop.screensaver"). Append the key name to this resource.

File: providers/screensaver.rb [hunk 6]
```diff
@@ -31,39 +31,31 @@
...
           value idle_enabled.to_s
-          schema "org.cinnamon.desktop.screensaver"
-          username username
-          provider "gecos_ws_mgmt_gsettings"
           action :nothing
         end.run_action(:set)
     
-        gecos_ws_mgmt_desktop_setting "lock-enabled" do
-          type "string"
+        desktop_gsettings "org.cinnamon.desktop.screensaver" do
```
Comment (alfem) [inline providers/screensaver.rb]: Avoid reusing a resource name ("org.cinnamon.desktop.screensaver"). Append the key name to this resource.

File: providers/screensaver.rb [hunk 7]
```diff
@@ -31,39 +31,31 @@
...
           value idle_delay
-          schema "org.cinnamon.desktop.session"
-          username username
-          provider "gecos_ws_mgmt_gsettings"
           action :nothing
         end.run_action(:set)
     
-        gecos_ws_mgmt_desktop_setting "lock-delay" do
-          type "string"
+        desktop_gsettings "org.cinnamon.desktop.screensaver" do
```
Comment (alfem) [inline providers/screensaver.rb]: Avoid reusing a resource name ("org.cinnamon.desktop.screensaver"). Append the key name to this resource.

**Comment by joroca** [comment]
Corrected. uniquely resources naming

File: metadata.rb [hunk 8]
```diff
@@ -8,6 +8,7 @@
 
 depends "apt"
 depends "chef-client"
+depends "compat_resource"
```
Comment (alfem) [inline metadata.rb]: This cookbook is deprecated

https://supermarket.chef.io/cookbooks/compat_resource

**Pull request merged**